### DISCUSSION TITLE
Return comments by username via get-posts-for-public-key API call

### DISCUSSION THREAD

**Discussion by brootle** [deso-protocol/dips#213] [Medium CIPs]
Add parameter like `"Comments": true` for `get-posts-for-public-key` API call.

Right now it is not possible to get comments (that are technically are posts that have parent post) made by user via DeSo API. Need to be able to fetch posts only, comments only, posts and comments together.

PS this is also super important for indexing posts and comments for DeSo search engines

**Comment by arhebbar** [comment]
While we are at it, is there a way we can get all the threaded comments of a main post in one go? Let the app layer figure out how to display it.

**Comment by darian-parrish** [comment]
This is already covered very similarly here: https://github.com/deso-protocol/dips/discussions/93
